pkg/reconciler/mutatingwebhook/reconcile.go

```diff
@@ -0,0 +1,82 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"k8s.io/api/admissionregistration/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/injector"
+	"github.com/openservicemesh/osm/pkg/logger"
+)
+
+var log = logger.New("reconciler")
+
+// MutatingWebhookConfigrationReconciler reconciles a MutatingWebhookConfiguration object
+type MutatingWebhookConfigrationReconciler struct {
+	client.Client
+	Scheme       *runtime.Scheme
+	OsmWebhook   string
+	OsmNamespace string
+	CertManager  certificate.Manager
+}
+
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=mutatingwebhookconfigurations,verbs=get;list;watch;update;
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=mutatingwebhookconfigurations/status,verbs=get;update;patch
+
+// Reconcile is the reconciliation method for OSM MutatingWebhookConfiguration.
+func (r *MutatingWebhookConfigrationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	instance := &v1beta1.MutatingWebhookConfiguration{}
+
+	if err := r.Get(ctx, req.NamespacedName, instance); err != nil {
+		log.Error().Err(err).Msgf("failure reading object %s ", req.NamespacedName.String())
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// reconcile only for OSM mutatingWebhookConfiguration
+	var shouldUpdate bool
+	if r.OsmWebhook == instance.Name {
```

Comment:
Should we check for if `req.NamespacedName` is OSM mutatingWebhookConfig at the beginning of this function?

Reply:
I thought a mutatingwebhookconfiguration is namespace independent and hence checking against the OSM namespace would be redundant for this particular scenario

Reply:
I was suggesting that we check the request mutatingwebhookconfig's name `req.NamespacedName` at the beginning of the function so that we can skip for other mutatingwebhookconfigs.

Reply:
okay makes sense, let me update this. I will use `req.Name` though, since there's no namespace for mutatingwebhookconfigurations

Reply:
SG! Both should be fine.